### PR TITLE
raise an exception on unhandled kwargs

### PR DESCRIPTION
**Pull request by chriddyp**

### CODE REVIEW

**Comment by chriddyp** [comment]
hey @theengineear @cldougl - an easy but important one. check out the changelog.md for details.

File: plotly/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '1.9.4'
+__version__ = '1.9.5'
```
Comment (cldougl) [inline plotly/version.py]: '1.9.6' srry @chriddyp just changed the version and updated pip

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,22 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+the `sharing` argument or mistakes the `sharing` argument for something else, like
+`privacy`, their mistake will be silently ignored and their graph will be made
+public:
+
+```
+py.plot(figure, sharig='private')
+```
+
+Now, an error will be raised.
+
```
Comment (cldougl) [inline CHANGELOG.md]: +1 for the reasoning

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,22 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+## [1.9.5] - 2015-01-15
+### Fixed
+Raise a `plotly.exceptions.PlotlyError` if the user supplied an unsupported
+keyword argument in `py.plot`.
+
+Here's why this matters. If a intends on making their graph private but misspells
```
Comment (theengineear) [inline CHANGELOG.md]: :goat: `If a intends`?

File: plotly/plotly/plotly.py [hunk 4]
```diff
@@ -209,6 +209,16 @@ def plot(figure_or_data, validate=True, **plot_options):
...
     """
+
+    for plot_option in plot_options:
+        if plot_option not in ['filename', 'fileopt', 'auto_open',
+                               'sharing', 'world_readable']:
+            raise PlotlyError(
+                'Unsupported keyword argument `' + plot_option + '`.\n' +
+                'Supported keyword arguments are: ' +
+                'filename, fileopt, auto_open, sharing, and '
+                'world_readable (deprecated)')
```
Comment (theengineear) [inline plotly/plotly/plotly.py]: :cow2: we usually use `format`.

File: plotly/plotly/plotly.py [hunk 5]
```diff
@@ -209,6 +209,16 @@ def plot(figure_or_data, validate=True, **plot_options):
                                      Make this figure private/public
 
     """
+
+    for plot_option in plot_options:
+        if plot_option not in ['filename', 'fileopt', 'auto_open',
+                               'sharing', 'world_readable']:
```
Comment (theengineear) [inline plotly/plotly/plotly.py]: :camel: Is this list used anywhere else? Should it be stored in a constant?

Comment (theengineear) [inline plotly/plotly/plotly.py]: Alternately, should we spell all these options out in the call signature?

File: plotly/tests/test_core/test_plotly/test_plot.py [hunk 6]
```diff
@@ -65,6 +65,10 @@ def test_plot_invalid_args_1(self):
     def test_plot_invalid_args_2(self):
         py.plot([1, 2, 3], [2, 1, 2], auto_open=False, filename='plot_invalid')
 
+    @raises(PlotlyError)
+    def test_plot_invalid_kwargs(self):
+        py.plot([{'x': [1, 2], 'y': [10, 20]}], privacy='private')
```
Comment (theengineear) [inline plotly/tests/test_core/test_plotly/test_plot.py]: thx for the :tiger2: !

**Comment by theengineear** [comment]
Non-blocking, but I think we're going to restrict the keyword args, we might as well just change the call signature, right?

:dancer:

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,22 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+## [1.9.5] - 2015-01-15
```
Comment (cldougl) [inline CHANGELOG.md]: `[1.9.6]` and `2016`

**Comment by theengineear** [comment]
@chriddyp should we close this one and try again in a new PR?

**Comment by chriddyp** [comment]
Yeah good idea. Still seems like a good idea, I forgot why I didn't end up merging this.

**Pull request closed without merging**